Tasks/UsePythonVersionV0/installpythonversion.ts

```diff
@@ -0,0 +1,133 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as semver from 'semver';
+import * as rest from 'typed-rest-client';
+import * as task from 'azure-pipelines-task-lib/task';
+import * as tool from 'azure-pipelines-tool-lib/tool';
+import * as osutil from './osutil';
+
+import { TaskParameters, PythonRelease, PythonFileInfo } from './interfaces';
+
+const MANIFEST_URL = 'https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json';
+const OS_VERSION = osutil._getOsVersion();
+
+/**
+ * Installs specified python version.
+ * This puts python binaries in the tools directory for later use.
+ * @param versionSpec version specification.
+ * @param parameters task parameters.
+ */
+export async function installPythonVersion(versionSpec: string, parameters: TaskParameters) {
+    const pythonInstallerDir: string = await downloadPythonVersion(versionSpec, parameters);
+
+    task.debug(`Extracted python archive to ${pythonInstallerDir}; running installation script`);
+
+    const installerScriptOptions = {
+        cwd: pythonInstallerDir,
+        windowsHide: true
+    };
+
+    if (os.platform() === 'win32') {
+        return task.exec('powershell', './setup.ps1', installerScriptOptions);
+    } else {
+        return task.exec('bash', './setup.sh', installerScriptOptions);
+    }
+}
+
+/**
+ * Downloads and extracts python file for the host system.
+ * Looks for python files from the github actions python versions manifest.
+ * Throws if file is not found.
+ * @param versionSpec version specification.
+ * @param parameters task parameters.
+ * @returns path to the extracted python archive.
+ */
+async function downloadPythonVersion(versionSpec: string, parameters: TaskParameters): Promise<string> {
+    if (!parameters.githubToken) {
+        throw new Error(task.loc('MissingGithubToken'));
```

Comment:
Why is this still requred?
Should there be some warning about possible limits hit?

Reply:
Made github token input optional. This line now outputs a warning instead of an error.